src/aics-image-viewer/components/TfEditor/index.tsx

```diff
@@ -356,24 +356,37 @@ export default class MyTfEditor extends React.Component<MyTfEditorProps, MyTfEdi
     svg.selectAll<SVGGElement, unknown>(".axis.axis--x").call(axis);
   }
 
+  private getBinLengths(): { binLengths: number[]; max: number } {
+    const histogram = this.props.channelData.histogram;
+    const binLengths = [];
+    let max = Number.NEGATIVE_INFINITY;
+    for (let i = 0; i < histogram.getNumBins(); i++) {
+      const binLength = histogram.getBin(i);
+      binLengths.push(binLength);
+      max = Math.max(max, binLength);
+    }
+    return { binLengths, max };
+  }
+
   // update the chart data
   private redrawHistogram(): void {
     d3.select(this.svgElement.current).select("g").selectAll(".bar").remove();
     if (this.props.volumeData && this.props.volumeData.length > 0) {
-      const bins = this.bins(this.props.volumeData);
-      this.binScale.domain([0.1, d3.max(bins, (d) => d.length)!]);
-      const bar = d3.select(this.svgElement.current).select("g").selectAll(".bar").data(bins);
+      const { binLengths, max } = this.getBinLengths();
+
+      this.binScale.domain([0.1, max]);
+      const bar = d3.select(this.svgElement.current).select("g").selectAll(".bar").data(binLengths);
       const barEnter = bar
         .enter()
         .append("g")
         .attr("class", "bar")
-        .attr("transform", (d) => `translate(${this.xScale(d.x0!)},${this.binScale(d.length)})`);
+        .attr("transform", (length, idx) => `translate(${this.xScale(idx)},${this.binScale(length)})`);
 
       barEnter
         .append("rect")
         .attr("x", 1)
-        .attr("width", (d) => d.x1! - d.x0!)
-        .attr("height", (d) => this.height - this.binScale(d.length));
+        .attr("width", () => 1)
```

Comment:
this function is awesome. Are you allowed to replace `()=>1` with `1` ?

Reply:
I am 😆